File: JsonSchemaAndStructs/KiteConnect-Kotlin-Structs/mf_sip_place.kt
// To parse the JSON, install Klaxon and do:
//
//   val mFSIPPlace = MFSIPPlace.fromJson(jsonString)

package quicktype

import com.beust.klaxon.*

private val klaxon = Klaxon()

data class MFSIPPlace (
    @Json(name = "\$ref")
    val ref: String,

    @Json(name = "\$schema")
    val schema: String,

    val definitions: Definitions
) {
    public fun toJson() = klaxon.toJsonString(this)

    companion object {
        public fun fromJson(json: String) = klaxon.parse<MFSIPPlace>(json)
    }
}

data class Definitions (
    @Json(name = "Data")
    val data: Data,

    @Json(name = "MFSIPPlace")
    val mfsipPlace: MFSIPPlaceClass
)

data class Data (
    val additionalProperties: Boolean,
    val properties: DataProperties,
    val required: List<String>,
    val title: String,
    val type: String
)

data class DataProperties (
    @Json(name = "sip_id")
    val sipID: Sipid
)

data class Sipid (
    val type: String
)

data class MFSIPPlaceClass (
    val additionalProperties: Boolean,
    val properties: MFSIPPlaceProperties,
    val required: List<String>,
    val title: String,
    val type: String
)

data class MFSIPPlaceProperties (
    val data: DataClass,
    val status: Sipid
)

data class DataClass (
    @Json(name = "\$ref")
    val ref: String
)

File: JsonSchemaAndStructs/KiteConnect-Rust-Structs/mf_sip_place.rs
// Example code that deserializes and serializes the model.
// extern crate serde;
// #[macro_use]
// extern crate serde_derive;
// extern crate serde_json;
//
// use generated_module::[object Object];
//
// fn main() {
//     let json = r#"{"answer": 42}"#;
//     let model: [object Object] = serde_json::from_str(&json).unwrap();
// }

extern crate serde_derive;

#[derive(Serialize, Deserialize)]
pub struct MfSipPlace {
    #[serde(rename = "$ref")]
    mf_sip_place_ref: String,

    #[serde(rename = "$schema")]
    schema: String,

    #[serde(rename = "definitions")]
    definitions: Definitions,
}

#[derive(Serialize, Deserialize)]
pub struct Definitions {
    #[serde(rename = "Data")]
    data: Data,

    #[serde(rename = "MFSIPPlace")]
    mfsip_place: MfsipPlaceClass,
}

#[derive(Serialize, Deserialize)]
pub struct Data {
    #[serde(rename = "additionalProperties")]
    additional_properties: bool,

    #[serde(rename = "properties")]
    properties: DataProperties,

    #[serde(rename = "required")]
    required: Vec<String>,

    #[serde(rename = "title")]
    title: String,

    #[serde(rename = "type")]
    data_type: String,
}

#[derive(Serialize, Deserialize)]
pub struct DataProperties {
    #[serde(rename = "sip_id")]
    sip_id: Sipid,
}

#[derive(Serialize, Deserialize)]
pub struct Sipid {
    #[serde(rename = "type")]
    sipid_type: String,
}

#[derive(Serialize, Deserialize)]
pub struct MfsipPlaceClass {
    #[serde(rename = "additionalProperties")]
    additional_properties: bool,

    #[serde(rename = "properties")]
    properties: MfsipPlaceProperties,

    #[serde(rename = "required")]
    required: Vec<String>,

    #[serde(rename = "title")]
    title: String,

    #[serde(rename = "type")]
    mfsip_place_class_type: String,
}

#[derive(Serialize, Deserialize)]
pub struct MfsipPlaceProperties {
    #[serde(rename = "data")]
    data: DataClass,

    #[serde(rename = "status")]
    status: Sipid,
}

#[derive(Serialize, Deserialize)]
pub struct DataClass {
    #[serde(rename = "$ref")]
    data_class_ref: String,
}
